conceptnet5/vectors/evaluation/analogy.py

```diff
@@ -165,8 +221,144 @@ def eval_analogies(frame):
+            question_pairs_scores.append(score)
+
+        our_answer_most = question_pairs[np.argmax(question_pairs_scores)]
+        our_answer_least = question_pairs[np.argmin(question_pairs_scores)]
+
+        if most == our_answer_most:
+            correct_most += 1
+        if least == our_answer_least:
+            correct_least += 1
+        total += 1
+
+    # Compute Spearman correlation of our ranks and MT ranks
+    our_semeval_scores = [score for pair, score in sorted(our_pair_scores.items())]
+    turk_semeval_scores = [score for pair, score in turk_rank]
+    spearman = round(spearmanr(our_semeval_scores, turk_semeval_scores)[0], 3)
+    spearman_results = confidence_interval(spearman, total)
+
+    # Compute an accuracy score on MaxDiff questions
+    maxdiff = round((correct_least + correct_most) / (total + total), 3)
+    low_maxdiff, high_maxdiff = proportion_confint((correct_least + correct_most), (2 * total))
+    maxdiff_results = pd.Series([maxdiff, low_maxdiff, high_maxdiff], index=['acc', 'low', 'high'])
+
+    return [maxdiff_results, spearman_results]
+
+
+def eval_semeval2012_global(vectors, weight_direct, weight_transpose, subset):
+    """
+    Return the average Spearman score and MaxDiff accuracy score for the entire test set.
+    """
+    spearman_scores = []
+    maxdiff_scores = []
+    for subclass in product(range(1, 11), 'a b c d e f g h i j'):
+        subclass = ''.join([str(element) for element in subclass])
+        try:
+            maxdiff, spearman = eval_semeval2012_analogies(vectors, weight_direct,
+                                                           weight_transpose,
+                                                           subset, subclass)
+            spearman_scores.append(spearman)
+            maxdiff_scores.append(maxdiff)
+        except FileNotFoundError:
+            continue
+
+    spearman_output = []
+    maxdiff_output = []
+    for interval in ['acc', 'low', 'high']:
+        average_maxdiff_score = np.mean([score[interval] for score in maxdiff_scores])
+        average_spearman_score = np.mean([score[interval] for score in spearman_scores])
+        spearman_output.append(average_spearman_score)
+        maxdiff_output.append(average_maxdiff_score)
+
+    return [pd.Series(maxdiff_output, index=['acc', 'low', 'high']),
+            pd.Series(spearman_output, index=['acc', 'low', 'high'])]
+
+
+def evaluate(frame, analogy_filename, subset='test', tune_analogies=True, semeval_scope='global'):
+    """
+    Run SAT and Semeval12-2 evaluations.
+    """
+    vectors = VectorSpaceWrapper(frame=frame)
+    results = empty_comparison_table()
+
+    if tune_analogies:
+        sat_weights = tune_pairwise_analogies(eval_pairwise_analogies, vectors, analogy_filename)
+        semeval_weights = tune_pairwise_analogies(eval_semeval2012_global, vectors)
+    else:
+        sat_weights = (0.35, 0.65)
+        semeval_weights = (0.2, 1.0)
```

Comment:
It looks like these came from the two ends of the scale that's currently in `tune_pairwise_analogies`, right? I'm trying it with a broader range of values.